pkg/pdatautil/hash.go

```diff
@@ -63,6 +63,26 @@ func MapHash(m pcommon.Map) [16]byte {
 	return hw.hashSum128()
 }
 
+// MapHashSelectedKeys return a hash for the provided map, using values of only provided keys.
+// Order of hash calculation is determined by the order of keys.
+func MapHashSelectedKeys(m pcommon.Map, keys []string) [16]byte {
```

Comment:
I'm wondering if it would make sense to filter out key value pairs that aren't in the `keys` input, and then call `MapHash` on the ones that are requested?

I'm thinking this would do the best job of reusing existing functionality, but happy to discuss.

Reply:
I agree it would be nice to reuse the functionality, but I figured it would be better to avoid unnecessary allocations, by just adding an additional method to the function. It also seem to fit pretty well in general lib contract.